### DISCUSSION TITLE
reducing energy consumption of moosefs in home setting

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#580] [Q&A]
Dear all

I understand that moosefs is an enterprise system and very strict standards about keeping our data safe
But desperate times - desperate measures. The energy prices do not justify running my moosefs cluster anymore.
And frankly, I don't want to move to another software.

I am trying to be creative and maybe lean on community experiences - what are the options to reduce power consumption for **home** cluster
currently my moosefs runs on bunch of single board ARM7 machines with single HDD each, 
And I love it, but it spins 24/7 and kills my electricity bills

I guess the options are
1. replace HDD with SSD (expensive and hard to achieve considering that my HDD range from 8 to 14TB)
2. reduce number of running HDD (possible, my cluster is only 60% full)
3. replace 7200 RPM with 5400 RPM (harder than I expected weirdly per TB the 7200 seems to be the cheapest option)
4. start switching the cluster off when not required. (I guess I could use something like network UPS signals to bring down the whole cluster in desired sequence)
5. move all running nodes to a single machine - I guess it would make option 4 much easier
 
I guess could also divide the cluster into two and use the one with reduced power as "colder" storage
this would make the "warmer" cluster much smaller (less hdd, less power, higher chances on energy efficient replacements)

But it would require some sort of management of tiered data to move it between clusters.

Is anyone out there having similar thoughts/challenges?
What else would you think about.

**Comment by tianon** [comment]
My home cluster has (as of the last few months) ended up on your option 5 -- all drives in a single machine, and I simply run a separate chunkserver process per drive (which frankly works pretty well).

Before this, I had effectively the same setup as now, but with the drives spread across two machines instead of one.

When it's ~idle, the power draw is much lower than the idle power draw of the two machines it used to run on (and that's with the single machine now running all the same processes/services as the previous two machines _plus_ the services of another unrelated machine I took down, so the machine _and_ MooseFS itself is seeing higher compute usage and still significantly lower power usage overall).

**Reply by eleaner** [reply]
thank you for the info
when you say idle - I believe there is no idle in moosefs - the integrity of the cluster is continuously checked in one massive loop.
What do you mean exactly? Do you mean when tere is no moosefs rebalancing? Or did you implement any dedicated power reduction measures?

**Reply by tianon** [reply]
Yeah sorry, that's not very precise -- I just mean "not doing much of anything".  When all _my_ workloads are idle (Plex server not in use, VMs all quiet, etc).  I haven't done anything special to actively reduce the amount of compute that MooseFS itself uses.

The load average of the system is just slightly above 1 when it's in the state I'm describing.

**Reply by struthio** [reply]
If You end up with one machine, why don't use some standard approaches like RAID ?
Isn't MooseFS a bit of overhead for managing different drives in one machine ?

**Reply by Lindsay-Mathieson** [reply]
> If You end up with one machine, why don't use some standard approaches like RAID ? Isn't MooseFS a bit of overhead for managing different drives in one machine ?

MooseFS can pool mixed sizes of drives. Most RAID setups require the drives to be the same size.

Also the fuse client can be used from other machines, which makes sharing the data across the network easy.

**Reply by struthio** [reply]
> MooseFS can pool mixed sizes of drives. Most RAID setups require the drives to be the same size.

Yes this is big advantage on MooseFS side. Some time ago I looked into it and for local machines combo of MergerFS + SnapRaid was used for such things. Now maybe there is something else - I am not tracking this topic.

> Also the fuse client can be used from other machines, which makes sharing the data across the network easy.

You could always use some other network filesystems like NFS, probably even with higher performance than MooseFS client.

And NFS now have clients for Linux as well as Windows machines.

--

Just to be clear. I am not arguing one way of another. Just had the same considerations as You some time ago - with final thought: If going for single machine why to use whole Distributed FS overhead and multiplying points of failure.

**Comment by struthio** [comment]
"But it would require some sort of management of tiered data to move it between clusters."

It depends if You are interested in 'safety' of data (ex. All data have at least two copies) or just combined disk space from different machines ?

In case of safety, then You could go with hot/cold storage, have good support for such approach with 'STRICT' storage classes and with division of CREATE/KEEP/ARCHIVE states.

In case of combined disk space, then probably one machine with some local RAID might be better solution.
